src/color.rs

```diff
@@ -0,0 +1,522 @@
+    /// <div style="background-color:new(0%, 0%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const BLACK: Self = Self::new(0.0, 0.0, 0.0, 1.0);
+    /// <div style="background-color:new(75%, 75%, 75%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const SILVER: Self = Self::new(0.75, 0.75, 0.75, 1.0);
+    /// <div style="background-color:new(50%, 50%, 50%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const GRAY: Self = Self::new(0.5, 0.5, 0.5, 1.0);
+    /// <div style="background-color:new(100%, 100%, 100%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const WHITE: Self = Self::new(1.0, 1.0, 1.0, 1.0);
+    /// <div style="background-color:new(50%, 0%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const MAROON: Self = Self::new(0.5, 0.0, 0.0, 1.0);
+    /// <div style="background-color:new(100%, 0%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const RED: Self = Self::new(1.0, 0.0, 0.0, 1.0);
+    /// <div style="background-color:new(50%, 0%, 50%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const PURPLE: Self = Self::new(0.5, 0.0, 0.5, 1.0);
+    /// <div style="background-color:new(100%, 0%, 100%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const FUCHSIA: Self = Self::new(1.0, 0.0, 1.0, 1.0);
+    /// <div style="background-color:new(0%, 50%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const GREEN: Self = Self::new(0.0, 0.5, 0.0, 1.0);
+    /// <div style="background-color:new(0%, 100%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const LIME: Self = Self::new(0.0, 1.0, 0.0, 1.0);
+    /// <div style="background-color:new(50%, 50%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const OLIVE: Self = Self::new(0.5, 0.5, 0.0, 1.0);
+    /// <div style="background-color:new(100%, 100%, 0%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const YELLOW: Self = Self::new(1.0, 1.0, 0.0, 1.0);
+    /// <div style="background-color:new(0%, 0%, 50%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const NAVI: Self = Self::new(0.0, 0.0, 0.5, 1.0);
+    /// <div style="background-color:new(0%, 0%, 100%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const BLUE: Self = Self::new(0.0, 0.0, 1.0, 1.0);
+    /// <div style="background-color:new(0%, 50%, 50%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const TEAL: Self = Self::new(0.0, 0.5, 0.5, 1.0);
+    /// <div style="background-color:new(0%, 100%, 100%); width: 10px; padding: 10px; border: 1px solid;"></div>
+    pub const AQUA: Self = Self::new(0.0, 1.0, 1.0, 1.0);
+
+    /// Creates a Color from new components
+    pub const fn new(r: f32, g: f32, b: f32, a: f32) -> Self {
+        Self { r, g, b, a }
+    }
+
+    /// Creates a Color from a hex value
+    pub fn from_hex(hex: u32) -> Self {
```

Comment:
why is alpha always 1.0?

Reply:
Because if no alpha is given, hex values mean to be fully visible.

Reply:
I added versions to also handle alpha values.

Reply:
lame